lib/widgets/message_list.dart

```diff
@@ -1104,10 +1107,15 @@ class StreamMessageRecipientHeader extends StatelessWidget {
               store.topicVisibilityPolicy(message.streamId, topic))),
         ]));
 
+    // When already in a topic narrow, disable tap interaction that would just
+    // push a MessageListPage for the same topic narrow.
+    // TODO(#1039) simplify by removing topic-narrow condition if we remove
+    //   recipient headers in topic narrows
     return GestureDetector(
-      onTap: () => Navigator.push(context,
-        MessageListPage.buildRoute(context: context,
-          narrow: TopicNarrow.ofMessage(message))),
+      onTap: !inTopicNarrow
```

Comment:
This comment is about the `onTap` parameter specifically, not the GestureDetector as a whole. So it should go immediately before the `onTap:` line.

In particular, the comment isn't saying anything about `onLongPress`, which we include regardless of this condition, nor about all the stuff in `child`.